--- a/builder/cc_builder.rs
+++ b/builder/cc_builder.rs
@@ -20,7 +20,7 @@ use crate::{
     cargo_env, emit_warning, env_var_to_bool, execute_command, find_clang_cl, get_crate_cc,
     get_crate_cflags, get_crate_cxx, is_no_asm, out_dir, requested_c_std, set_env_for_target,
-    should_build_jitter_entropy, target, target_arch, target_env, target_os, target_vendor,
+    set_env, should_build_jitter_entropy, target, target_arch, target_env, target_os, target_vendor,
     CStdRequested, EnvGuard, OutputLibType,
 };
@@ -363,8 +363,16 @@ impl CcBuilder {
     }
 
     pub fn prepare_builder(&self) -> cc::Build {
-        if let Some(cflags) = get_crate_cflags() {
-            set_env_for_target("CFLAGS", cflags);
+        if let Some(mut cflags) = get_crate_cflags() {
+            if target_vendor() == "apple" {
+                cflags = cflags
+                    .split_whitespace()
+                    .filter(|flag| !flag.starts_with("-mmacosx-version-min="))
+                    .collect::<Vec<_>>()
+                    .join(" ");
+                set_env("CFLAGS", &cflags);
+            }
+            set_env_for_target("CFLAGS", &cflags);
         }
 
         let mut cc_build = self.create_builder();
